File: csharp-supplychain/vulnerable9.cs
using Microsoft.ML.OnnxRuntime;

public class ModelLoader
{
    private readonly HttpClient _httpClient;

    public ModelLoader(HttpClient httpClient) => _httpClient = httpClient;

    public async Task<InferenceSession> LoadFromEnv()
    {
        var modelUrl = Environment.GetEnvironmentVariable("MODEL_DOWNLOAD_URL");
        var modelBytes = await _httpClient.GetByteArrayAsync(modelUrl);
        return new InferenceSession(modelBytes);
    }
}

File: java-supplychain/vulnerable9.java
import ai.onnxruntime.OrtEnvironment;
import ai.onnxruntime.OrtSession;
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;

public class ModelLoader {
    public OrtSession loadFromEnv() throws Exception {
        OrtEnvironment env = OrtEnvironment.getEnvironment();
        String modelUrl = System.getenv("MODEL_DOWNLOAD_URL");
        HttpClient client = HttpClient.newHttpClient();
        HttpRequest request = HttpRequest.newBuilder().uri(URI.create(modelUrl)).build();
        byte[] modelBytes = client.send(request, HttpResponse.BodyHandlers.ofByteArray()).body();
        return env.createSession(modelBytes);
    }
}
